MHIdentityKitTests/Infrastructure/OAuth2IdentityManagerAsyncTests.swift

```diff
@@ -0,0 +1,230 @@
+                }
+                else if callCount == 2 {
+                    
+                    //simulate valid token
+                    handler(AccessTokenResponse(accessToken: "tat2", tokenType: "Bearer", expiresIn: 1234, refreshToken: "trt2", scope: nil), nil)
+                }
+                else {
+                    
+                    handler(nil, ErrorResponse(code: .invalidGrant))
+                }
+            }
+            
+            func authenticateAsync() async throws -> AccessTokenResponse? {
+                
+                e.fulfill()
+                callCount += 1
+                
+                if callCount == 1 {
+                    
+                    //simulate expired token
+                    return AccessTokenResponse(accessToken: "tat1", tokenType: "Bearer", expiresIn: 0, refreshToken: "trt1", scope: nil)
+                }
+                else if callCount == 2 {
+                    
+                    //simulate valid token
+                    return AccessTokenResponse(accessToken: "tat2", tokenType: "Bearer", expiresIn: 1234, refreshToken: "trt2", scope: nil)
+                }
+                else {
+                    
+                    throw ErrorResponse(code: .invalidGrant)
+                }
+            }
+            
+        }
+        
+        class Refresher: AccessTokenRefresher {
+            
+            let e: XCTestExpectation
+            private var callCount = 0
+            
+            init(e: XCTestExpectation) {
+                
+                self.e = e
+            }
+            
+            func refresh(using requestModel: AccessTokenRefreshRequest, handler: @escaping (AccessTokenResponse?, Error?) -> Void) {
+                
+                e.fulfill()
+                callCount += 1
+                
+                if callCount == 1 {
+                    
+                    XCTAssertEqual(requestModel.refreshToken, "trt1")
+                    handler(AccessTokenResponse(accessToken: "rtat", tokenType: "Bearer", expiresIn: 0, refreshToken: "trt3", scope: nil), nil)
+                }
+                else {
+                    
+                    XCTAssertEqual(requestModel.refreshToken, "trt3")
+                    handler(nil, MHIdentityKitError.authenticationFailed(reason: MHIdentityKitError(error: ErrorResponse(code: .invalidGrant))))
+                }
+            }
+        }
+        
+        let manager = OAuth2IdentityManager(flow: Flow(e: XCTestExpectation(description: "XCTestCase Default Expectation")), refresher: Refresher(e: XCTestExpectation(description: "XCTestCase Default Expectation")), storage: InMemoryIdentityStorage(), authorizationMethod: .header)
+        
+        let request1 = try! await manager.authorizeAsync(request: URLRequest(url: URL(string: "http://foo.bar")!), forceAuthenticate: false)
```

Comment:
Make the test function throwing, so there will be no need to force unwrap the error. 

This should be applied to all similar cases.